src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -154,11 +156,15 @@ public StreamingResponseBody streamEvents(
                         streamTimeout, streamKeepAliveLimit, maxUncommittedSize,
                         nakadiSettings.getDefaultCommitTimeoutSeconds(), client.getClientId());
                 streamer = subscriptionStreamerFactory.build(subscriptionId, streamParameters, output,
-                        connectionReady, blacklistService);
+                        connectionReady, blacklistService, client);
                 streamer.stream();
             } catch (final InterruptedException ex) {
                 LOG.warn("Interrupted while streaming with " + streamer, ex);
                 Thread.currentThread().interrupt();
+            } catch (final AccessDeniedException e) {
+                writeProblemResponse(response, output.getOutputStream(), Problem.valueOf(Response.Status.FORBIDDEN,
```

Comment:
maybe it is better to put this logic in SubscriptionOutputImpl.onException it will be simpler to refactor exceptions later